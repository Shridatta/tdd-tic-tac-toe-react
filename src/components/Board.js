import React, { Component } from "react";
import "./App.css";
import Square from "./Square";

class Board extends Component {
  constructor(props) {
    super(props);
    this.state = {
      squares: Array(9).fill(null),
      xTurnToPlay: true
    };
    console.log(this.state);
  }

  renderSlot(i) {
    return <Square />;
  }
  render() {
    return (
      <div className="board">
        Board <br />
        <div className="currentPlayer">The current player is</div>
        <div className="row" />
        <div className="row">Row 2</div>
        <div className="row">Row 3</div>
      </div>
    );
  }
}

export default Board;
