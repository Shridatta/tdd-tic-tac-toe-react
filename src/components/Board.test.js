import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Board from "./Board";

configure({ adapter: new Adapter() });

const board = shallow(<Board />);

describe("when rendering the app", () => {
  it("renders Board correctly", () => {
    expect(board).toMatchSnapshot();
  });

  it("initializes the board state with null", () => {
    expect(board.state().squares).toEqual([
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null
    ]);
  });
  it("initializes the state with X's turn to play to true", () => {
    expect(board.state().xTurnToPlay).toEqual(true);
  });

  it("should render the the Current Player div", () => {
    expect(board.find(".currentPlayer").exists()).toBeTruthy();
  });
});
