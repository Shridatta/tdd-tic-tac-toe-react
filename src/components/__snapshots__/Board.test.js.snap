// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering the app renders Board correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Board ",
        <br />,
        <div
          className="currentPlayer"
        >
          The current player is
        </div>,
        <div
          className="row"
        >
          <Square
            className="squareinit"
            slotNumber={1}
            value={null}
          />
        </div>,
        <div
          className="row"
        >
          Row 2
        </div>,
        <div
          className="row"
        >
          Row 3
        </div>,
      ],
      "className": "board",
    },
    "ref": null,
    "rendered": Array [
      "Board ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The current player is",
          "className": "currentPlayer",
        },
        "ref": null,
        "rendered": "The current player is",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Square
            className="squareinit"
            slotNumber={1}
            value={null}
          />,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "squareinit",
            "slotNumber": 1,
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Row 2",
          "className": "row",
        },
        "ref": null,
        "rendered": "Row 2",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Row 3",
          "className": "row",
        },
        "ref": null,
        "rendered": "Row 3",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Board ",
          <br />,
          <div
            className="currentPlayer"
          >
            The current player is
          </div>,
          <div
            className="row"
          >
            <Square
              className="squareinit"
              slotNumber={1}
              value={null}
            />
          </div>,
          <div
            className="row"
          >
            Row 2
          </div>,
          <div
            className="row"
          >
            Row 3
          </div>,
        ],
        "className": "board",
      },
      "ref": null,
      "rendered": Array [
        "Board ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The current player is",
            "className": "currentPlayer",
          },
          "ref": null,
          "rendered": "The current player is",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Square
              className="squareinit"
              slotNumber={1}
              value={null}
            />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "squareinit",
              "slotNumber": 1,
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Row 2",
            "className": "row",
          },
          "ref": null,
          "rendered": "Row 2",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Row 3",
            "className": "row",
          },
          "ref": null,
          "rendered": "Row 3",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
